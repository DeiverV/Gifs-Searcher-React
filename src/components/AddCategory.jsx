import PropTypes from "prop-types";
import { useState } from "react"

export const AddCategory = ({onAddCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        
        const newCategorieValue = inputValue.trim();
        if(newCategorieValue.length<=1) return;

        onAddCategory(newCategorieValue);
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input 
                type='text' 
                placeholder="Buscar Gifs" 
                value={inputValue} 
                onChange={onInputChange} 
            />
        </form>
    )

}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired,
}
