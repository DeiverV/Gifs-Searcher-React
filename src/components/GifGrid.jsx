import { GifItem, LoadingMessage } from "./";

import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({category, onDeleteCategory}) => {

    const {images, isLoading} = useFetchGifs(category);

    const deleteCategory = (event) => onDeleteCategory(category);

    return (
        <>
            <h3>{ category }</h3>
            <LoadingMessage isLoading={isLoading}/>
            <button onClick={deleteCategory}>delete</button>
            <div className="card-grid">
                {
                    images.map((image) => (
                        <GifItem key={image.id} {...image}/>
                    ))
                }
            </div>
        </>
    )

}
