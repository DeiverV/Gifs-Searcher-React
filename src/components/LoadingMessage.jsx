export const LoadingMessage = ({isLoading}) => {
  return (
    isLoading && <h3>Loading...</h3>
  )
}
