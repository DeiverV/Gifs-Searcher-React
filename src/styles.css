* {
    font-family: Helvetica, Arial, sans-serif;
    background-color: var(--color4);
    color: var(--color2);
  }

  :root{
    --color1: #001600;
    --color2:#004536;
    --color3:#ff6f60;
    --color4:#ffb280;
    --color5: #ffeeb0;
  }
  
  body {
    padding: 30px;
  }
  
  input {
    background-color: var(--color1);
    color: var(--color4);
    border-radius: 5px;
    border: 1px solid rgb(97, 32, 158);
    font-size: 1.3rem;
    outline: none;
    padding: 10px 15px;
    width: 90%;
  }

  button{
    background-color: var(--color2);
    color: var(--color4);
    font-size: 1.3rem;
    padding: 10px;
    border: none;
    border-radius: 20px;
    box-shadow: 0 0 5px 1px var(--color1);
    margin-bottom: 20px;
    width: 100%;
    cursor: pointer;
    text-transform: uppercase;
    transition: .2s;
  }

  button:hover {
    background-color: #001600;
  }

  h1{
    font-size: 3rem;
    font-style: italic;
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
    color: var(--color1);
  }
  
  h2 {
    font-size: 1.5rem;
  }
  
  h3 {
    font-size: 3rem;
    color: var(--color1);
    margin-bottom: 25px;
    width: 100%;
    border-bottom: 2px solid var(--color1);
  }

  .card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }
  
  .card {
    align-content: center;
    align-items: center;
    background-color: var(--color2);
    border-radius: 10px;
    box-shadow: 0px 0px 20px 5px rgba(0, 0, 0, 0.308);
    display: flex;
    flex-direction: column;
    height: 0%;
    justify-content: center;
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;
  }
  
  .card p {
    background-color: #004536;
    color: var(--color4);
    flex: 1;
    font-size: 1.1rem;
    padding: 5px 20px 0px 20px;
    text-align: center;
    text-transform: capitalize;
  }
  
  .card img {
    width: 100%;
  }
  /*-----------------------Loader*/
  .lds-circle {
    display: inline-block;
    transform: translateZ(1px);
  }
  .lds-circle > div {
    display: inline-block;
    width: 64px;
    height: 64px;
    margin: 8px;
    border-radius: 50%;
    background: #fff;
    animation: lds-circle 2.4s cubic-bezier(0, 0.2, 0.8, 1) infinite;
  }
  @keyframes lds-circle {
    0%, 100% {
      animation-timing-function: cubic-bezier(0.5, 0, 1, 0.5);
    }
    0% {
      transform: rotateY(0deg);
    }
    50% {
      transform: rotateY(1800deg);
      animation-timing-function: cubic-bezier(0, 0.5, 0.5, 1);
    }
    100% {
      transform: rotateY(3600deg);
    }
  }
  