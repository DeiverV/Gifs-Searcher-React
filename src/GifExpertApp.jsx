//API Key "NGlhJZzxWSO2n4zLT9wfEC8kRYi22WRr"

import { useState } from "react"
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState([ 'One Punch','Naruto' ]);

    const onAddCategory = (newCategorie) =>{

        if( categories.includes(newCategorie) ) return;

        // setCategories(categories.concat('Lol'))
        setCategories([newCategorie ,...categories])
    } 

  return (
    <>
        {/* Titulo */}
        <h1>GifExpertApp</h1>

        {/* Input */}
        {/* <AddCategory setCategories={ setCategories } categories={categories} /> */}
        <AddCategory onAddCategory={ onAddCategory } />
        {/* Listado de Gif*/}
        <ol>
            { categories.map( category => {
                return <li key={ category }>{ category }</li>
            })}
        </ol>
            {/* Gif Item */}

    </>

  )
}
