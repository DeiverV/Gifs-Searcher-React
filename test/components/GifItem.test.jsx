import { fireEvent, render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';

describe('Test in GifItem Component', () => {
    
    const title = 'Dragon ball gif';
    const url = 'http://wtfmomento.com/hola.jpg';
    
    test('should render a gif', () => { 
        const { container } = render(<GifItem title={title} url={url}/>);
        expect( container ).toMatchSnapshot();
    })

    test('debe mostrar la imagen con url y alt indicado', () => { 
        render(<GifItem title={title} url={url} />);
        //screen.debug()
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const  {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title)
    })

    test('debe de mostrar el titulo del componente', () => { 
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
})