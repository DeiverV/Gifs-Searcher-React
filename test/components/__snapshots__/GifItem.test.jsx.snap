// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifItem Component should render a gif 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon ball gif"
      src="http://wtfmomento.com/hola.jpg"
    />
    <p>
      Dragon ball gif
    </p>
  </div>
</div>
`;
